kernel/src/introspection.rs

```diff
@@ -0,0 +1,107 @@
+//! Mechanism for inspecting the status of the kernel.
+//!
+//! In particular this provides functions for getting the status of processes
+//! on the board. It potentially could be expanded to other kernel state.
+
+use core::cell::Cell;
+
+use callback::AppId;
+use common::cells::NumericCellExt;
+use process;
+use sched::Kernel;
+
+/// This struct provides the introspection functions.
+pub struct Introspection {
```

Comment:
I don't think Introspection is quite the right name here: in libraries, that usually allows something to inspect itself, or look inside its internal structure/metadata. I might call it ProcessInfo.